easybuild/easyblocks/p/parmetis.py

```diff
@@ -166,20 +167,33 @@ def install_step(self):
         # other applications depending on ParMETIS (SuiteSparse for one) look for both ParMETIS libraries
         # and header files in the Lib directory (capital L). The following symlink are hence created.
         try:
-            llibdir = os.path.join(self.installdir, 'Lib')
-            os.symlink(libdir, llibdir)
-            for f in ['metis.h', 'parmetis.h']:
-                os.symlink(os.path.join(includedir, f), os.path.join(libdir, f))
+            caplibdir = os.path.join(self.installdir, 'Lib')
+            if os.path.lexists(caplibdir):
+                remove_file(caplibdir)
+            symlink(libdir, caplibdir)
+            for header_file in ['metis.h', 'parmetis.h']:
+                header_path = os.path.join(libdir, header_file)
+                if os.path.lexists(header_path):
+                    remove_file(header_path)
+                symlink(os.path.join(includedir, header_file), header_path)
         except OSError as err:
             raise EasyBuildError("Something went wrong during symlink creation: %s", err)
 
     def sanity_check_step(self):
         """Custom sanity check for ParMETIS."""
 
+        # Paths to libraries defined depending on SHARED configuration option
+        config_shared = ['-DSHARED=%s' % val for val in ['1', 'ON', 'YES', 'TRUE', 'Y']]  # True values in CMake
```

Comment:
@lexming there are many more valid ways to specify this cmake option, eg. `-DSHARED='1'`, `-D SHARED=1`, `-DSHARED:BOOL=ON`, and so on.

to make this more robust, it's probably better to create a `sharedlibs` parameter, like is done for example in the ATLAS easyblock.

Reply:
I thought about using `sharedibs` as an extra option, but it is not trivial for `parmetis` as it is necessary to run multiple iterations to have an installation with both static and shared libs. So, to still allow for such builds, I improved the detection of the `SHARED` option in `configopts` to be more flexible and work for both lists and strings.